### DISCUSSION TITLE
There should be an option so that runtime errors don't crash the whole page

### DISCUSSION THREAD

**Discussion by dscafati** [vercel/next.js#24099] [Ideas]
### Describe the feature you'd like to request

I noticed that simple uncaught typos or syntax errors can make the whole website crash showing the error page. Most of the time, showing this error page is worse than just showing the page with a few broken features. This is specially important when you are paying for getting traffic and they all see an error page just because of a silly runtime error. 

### Describe the solution you'd like

I would like a way to disable this behavior. In case of errors, the developer should decide to continue to show the broken page even if some functionality throws an error, or to hide everything and show the error page.

### Describe alternatives you've considered

I couldn't find any alternative that works on my exported static site, it seems to be no way to overwrite this behavior

**Comment by akellbl4** [comment]
Hi,
You can catch errors on any level of your application in JSX with [Error Boundary](https://reactjs.org/docs/error-boundaries.html) and show fallback for that place.
For fetching methods you can use just `try {...} catch (e) {...}`

Also, I could suggest using any sort of error logging system such as Sentry. With it you will see errors and it will be easier to fix :)

**Reply by dscafati** [reply]
Thanks for the reply. Is there a way to capture errors with error boundaries, but continue to show the already rendered subtree? That's what I'm trying to achieve.

**Reply by dscafati** [reply]
In other words, if I wrap the app with an error boundary component I have to either show an error page that overwrites the rendered content, or a blank page.

**Reply by akellbl4** [reply]
When you have an error inside of a component you have no change to render it because that how JS works.
You can wrap small parts of your interface and you will have a major part of the page rendered and only a small part that was broken won't be rendered.

**Reply by dscafati** [reply]
Ok, I think I'm looking for a more general solution at nextjs level. Like if the page crashes, the whole hydrate process is ignored and the server side rendered page continues to be shown. I'll wait to see if others reply or this can be converted back into a feature request

**Reply by akellbl4** [reply]
As far as I know you have no other options to do so.
